backend/msgpack/writer.go

```diff
@@ -156,11 +156,10 @@ func (w *writer) flushWithLock() error {
 			encoder.Unlock()
 			continue
 		}
-		newBufferedEncoder := w.encoderPool.Get()
-		newBufferedEncoder.Reset()
-		encoder.Reset(newBufferedEncoder)
+		// Use a sizeBefore of 0 to indicate we want to write the entire buffer.
+		oldBufferedEncoder := w.prepareEnqueueBufferWithLock(encoder, 0)
```

Comment:
Hm this seems like a bit of a hack, perhaps we should leave it as is just to be very explicit about writing out the whole buffer.